contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -3695,43 +3694,55 @@ exec_stmt_grantschema(PLtsql_execstate *estate, PLtsql_stmt_grantschema *stmt)
 	 * sysadmin or login is not the schema owner, then it doesn't have the permission to GRANT/REVOKE.
 	 */
 	login_is_db_owner = 0 == strncmp(login, get_owner_of_db(dbname), NAMEDATALEN);
-	datdba = get_role_oid("sysadmin", false);
 	schema_name = get_physical_schema_name(dbname, stmt->schema_name);
-	schemaOid = LookupExplicitNamespace(schema_name, true);
+
+	if(schema_name)
+	{
+		schemaOid = LookupExplicitNamespace(schema_name, true);
+	}
+	else
+	{
+		ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_SCHEMA),
+					 errmsg("An object or column name is missing or empty. For SELECT INTO statements, verify each column has a name. For other statements, look for empty alias names. Aliases defined as \"\" or [] are not allowed. Change the alias to a valid name.")));
```

Comment:
Error code(although right) seems to have an ambiguous relation with error message. So Why have we c/p sql server error message here?

Reply:
Shouldn't we have the same message as SQL Server?
What do you suggest?

Reply:
Message from Rob:

```
Same as SQL Server, if possible, would always be best.
 
Exception: when a part of the error msg makes no sense for Babelfish , like saying you cannot change ownership for 'master, model and tempdb databases' – since we don't supporet model, that was removed from the error msg. But we don't seem to have such a situation here